### PR TITLE
feat: subsection list

### PR DESCRIPTION
**Pull request by navinkarkera**
**Description:** This PR adds subsection component which enables UI to display subsections under each section. It also adds functionality for adding new subsections, duplicating, deleting and publishing subsections. We also try to reuse section list as much as possible.

`Private-ref`: https://tasks.opencraft.com/browse/BB-8221

**Dependents on**: 
* https://github.com/openedx/edx-platform/pull/33667
* https://github.com/openedx/frontend-app-course-authoring/pull/694

**Testing instructions**:

* Setup this repo as described in the readme.
* Checkout branch from https://github.com/openedx/edx-platform/pull/33667 and run `make cms-up` in devstack.
* Open any course outline by using this url format: http://localhost:2001/course/course-v1:edX+DemoX+Demo_Course/

**Screenshot:**

image

### CODE REVIEW

File: src/course-outline/hooks.jsx [hunk 1]
```diff
@@ -79,34 +91,54 @@ const useCourseOutline = ({ courseId }) => {
...
+      break;
+    case COURSE_BLOCK_NAMES.sequential.id:
+      dispatch(deleteCourseSubsectionQuery(currentItem.id, currentSection.id));
+      break;
+    case COURSE_BLOCK_NAMES.vertical.id:
+      dispatch(deleteCourseSubsectionQuery(
+        currentItem.id,
+        currentSubsection.id,
+        currentSection.id,
+      ));
```
Comment (CefBoud) [inline src/course-outline/hooks.jsx]: Is this in prevision of the Unit implementation? Should the deletion happen through `deleteCourseSubsectionQuery`?

File: src/course-outline/highlights-modal/HighlightsModal.jsx [hunk 2]
```diff
@@ -23,7 +23,7 @@ const HighlightsModal = ({
   onSubmit,
 }) => {
   const intl = useIntl();
-  const { highlights = [], displayName } = useSelector(getCurrentSection);
+  const { highlights = [], displayName } = useSelector(getCurrentItem);
   const initialFormValues = getHighlightsFormValues(highlights);
```
Comment (CefBoud) [inline src/course-outline/highlights-modal/HighlightsModal.jsx]: Nit: please correct me if I am wrong, but highlights are specific to sections. Any reason to switch to `getCurrentItem` ?

File: src/course-outline/hooks.jsx (hunk 1 above)
Comment (navinkarkera) [reply src/course-outline/hooks.jsx]: @CefBoud Nice catch! Missed to update it after copying.

File: src/course-outline/highlights-modal/HighlightsModal.jsx (hunk 2 above)
Comment (navinkarkera) [reply src/course-outline/highlights-modal/HighlightsModal.jsx]: No, actually `getCurrentSection` is better. Its just that I added `getCurrentSection` back at a later point for delete option.

**Review by CefBoud — APPROVED** [review]
Great job on this, @navinkarkera !

**Pull request merged**